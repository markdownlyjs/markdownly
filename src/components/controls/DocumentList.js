import React from 'react';
import PropTypes from 'prop-types';
import DocumentForm from './DocumentForm';

const styles = {
  section: {
    padding: 10
  },
  list: {
    listStyle: 'none',
    display: 'flex',
    padding: 0
  }
};

const DocumentList = ({ documents, inputValue, handleChange, handleFormSubmit, handleSelect }) => {

  const documentList = documents.map((doc, i) => (
    <li key={i}>
      <button onClick={() => handleSelect(i)}>{doc.title}</button>
    </li>
  ));

  return (
    <section style={styles.section}>
      <ul style={styles.list}>
        {documentList}
      </ul>
      <DocumentForm 
        inputValue={inputValue} 
        handleChange={handleChange} 
        handleSubmit={handleFormSubmit}
      />
    </section> 
  );
};

DocumentList.propTypes = { 
  documents: PropTypes.array.isRequired,
  inputValue: PropTypes.string.isRequired,
  handleFormSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSelect: PropTypes.func.isRequired,
};

export default DocumentList;
