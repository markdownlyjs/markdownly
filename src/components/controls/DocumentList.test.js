import React from 'react';
import { shallow } from 'enzyme';
import DocumentList from './DocumentList';

describe('DocumentList component', () => {
  it('renders DocumentList', () => {
    const wrapper = shallow(<DocumentList 
      documents={[]}
      handleFormSubmit={() => {}} 
      handleChange={() => {}}
      inputValue=''
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
