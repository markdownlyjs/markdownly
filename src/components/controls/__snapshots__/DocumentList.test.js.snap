// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentList component renders DocumentList 1`] = `
<ul>
  <li>
    <Connect(AddDocument) />
  </li>
</ul>
`;
